src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -463,6 +507,88 @@ private void clearConfig() {
     setDefaultValues();
   }
 
+  private boolean checkIfNeedsSaving() {
+    boolean needsSaving = true;
+    try {
+      String currentConfigString =
+          new ObjectMapper()
+              .writer()
+              .withDefaultPrettyPrinter()
+              .writeValueAsString(createRawContestConfig());
+      if (selectedFile == null && currentConfigString.equals(emptyConfigString)) {
+        // All fields are currently empty / default values so no point in asking to save
+        needsSaving = false;
+      } else if (GuiContext.getInstance().getConfig() != null) {
+        String savedConfigString =
+            new ObjectMapper()
+                .writer()
+                .withDefaultPrettyPrinter()
+                .writeValueAsString(GuiContext.getInstance().getConfig().rawConfig);
+        needsSaving = !currentConfigString.equals(savedConfigString);
+      }
+    } catch (JsonProcessingException exception) {
+      Logger.log(
+          Level.WARNING,
+          "Unable tell if saving is necessary, but everything should work fine anyway! Prompting for save just in case...\n%s",
+          exception.toString());
+    }
+    return needsSaving;
+  }
+
+  private boolean checkForSaveAndContinue() {
```

Comment:
If I'm reading this method correctly... the only way you can end up with willContinue = false is if the user clicks "Save" but then doesn't actually save?

Reply:
Exactly, if they close the window or press cancel. I think it's easier to check this condition the way I wrote it, rather than trying to capture those other cases.